python/tskit/trees.py

```diff
@@ -3446,14 +3446,20 @@ def edge_diffs(self):
         The iterator yields a sequence of 3-tuples, ``(interval, edges_out,
         edges_in)``. The ``interval`` is a pair ``(left, right)`` representing
         the genomic interval (see :attr:`Tree.interval`). The ``edges_out``
-        value is a tuple of the edges that were just-removed to create the tree
+        value is a list of the edges that were just-removed to create the tree
         covering the interval (hence ``edges_out`` will always be empty for the
-        first tree). The ``edges_in`` value is a tuple of edges that were just
+        first tree). The ``edges_in`` value is a list of edges that were just
         inserted to construct the tree convering the current interval. The
-        iterator terminates after the final interval in the tree sequence (i.e.
-        it does not report a final removal of all remaining edges); the number
-        of iterations is thus always equal to the number of trees in the tree
-        sequence.
+        edges in each ``edges_in`` list are returned in the
+        :ref:`order required for a valid tree sequence <sec_edge_requirements>`
```

Comment:
I don't think this is true - we're not returning edges in the table order, it's the order given by the indexes. I think it's also not in order of ID, it's in order of parent node time.

Reply:
Ah - I wasn't exactly sure. But are the edges for a given parent node always going to be returned consecutively?

Reply:
(NB: @jeromekelleher - this is important for detecting polytomies duding edge diff iteration)

Reply:
Here's where it's done: https://github.com/tskit-dev/tskit/blob/master/c/tskit/tables.c#L7203

So it looks like the sort is ``(left, time[parent], parent, child)``. This does guarantee that all edges for a given parent are adjacent, and children are ordered by node ID.

Reply:
Thanks. I'll adjust my wording. I think the inverse order is used for edges_out?

Reply:
Yes, it's the same order but reversed.